common/src/main/java/com/microsoft/identity/common/adal/internal/util/StringExtensions.java

```diff
@@ -247,4 +254,74 @@ public static boolean hasPrefixInHeader(final String value, final String prefix)
     public static String base64UrlEncodeToString(final String message) {
         return Base64.encodeToString(message.getBytes(Charset.forName(ENCODING_UTF8)), Base64.URL_SAFE | Base64.NO_WRAP);
     }
+
+
+
+    /**
+     * Decode the given url, and convert it into map with the given delimiter.
```

Comment:
I'd like to see this method removed entirely (or implemented differently).

IF it stays, this javadoc should be improved to say what it places in the `Map`, specifically the query params